### PR TITLE
7pm 2 add navbar to storybook

### PR DESCRIPTION
**Pull request by thiskappaisgrey**
This PR adds the `AppNavbar` storybook. 

In order to complete this assignment, we refactored the visual elements of the nav bar into a function component(I call it a "pure" component, evoking "pure" functions, not evoking the react PureComponent thing) and rendering that using the storybook. This way we don't have to do any weird mocking swr to render the different states of the navbar.

### CODE REVIEW

**Review by BaldPulse — DISMISSED** [review]
Looks good to me.

File: javascript/src/main/components/Nav/AppNavbar.js [hunk 1]
```diff
@@ -7,6 +7,44 @@ import { useAuth0 } from "@auth0/auth0-react";
 import { fetchWithToken } from "main/utils/fetch";
 import NavbarHover from "../Nav/NavbarHover";
 
+export function PureNavbar({ isAdmin, isMember, ChannelPages, AdminPages, SearchPages, DataPages }) {
```
Comment (btk5h) [inline javascript/src/main/components/Nav/AppNavbar.js:10]: By convention, prop names are usually lowerCamelCase. Can you rename this?

File: javascript/src/main/components/Nav/AppNavbar.js [hunk 2]
```diff
@@ -7,6 +7,44 @@ import { useAuth0 } from "@auth0/auth0-react";
...
+  return (
+    <Navbar bg="dark" variant="dark" expand="lg">
+      <Navbar.Toggle />
+      <Navbar.Collapse>
+        <LinkContainer to={""}>
```
Comment (btk5h) [inline javascript/src/main/components/Nav/AppNavbar.js:15]: While you're refactoring these components, could you also clean up cases of `prop={"literal"}`, which can be replaced with `prop="literal"`?

**Review by btk5h — CHANGES_REQUESTED** [review]

File: javascript/src/main/components/Nav/AppNavbar.js (hunk 1 above)
Comment (thiskappaisgrey) [reply javascript/src/main/components/Nav/AppNavbar.js]: Done!

File: javascript/src/main/components/Nav/AppNavbar.js (hunk 2 above)
Comment (thiskappaisgrey) [reply javascript/src/main/components/Nav/AppNavbar.js:15]: Done!

**Review by jtamayo0217 — DISMISSED** [review]
LGTM

**Review by btk5h — APPROVED** [review]

**Review by jtamayo0217 — APPROVED** [review]
Looks good to me!

**Pull request merged**